packages/govuk-frontend/src/govuk/init.jsdom.test.mjs

```diff
@@ -292,6 +294,49 @@ describe('createAll', () => {
     )
   })
 
+  it('executes overloaded checkSupport of component', () => {
+    const componentRoot = document.createElement('div')
+    componentRoot.setAttribute('data-module', 'mock-component')
+    document.body.appendChild(componentRoot)
+
+    const checkSupportMock = jest.fn(() => {})
+
+    class MockComponentWithCheckSupport extends MockComponent {
+      checkSupport() {
+        checkSupportMock()
+      }
+    }
+
+    const result = createAll(MockComponentWithCheckSupport)
+
+    expect(checkSupportMock).toHaveBeenCalled()
+    expect(result).toStrictEqual([expect.any(MockComponent)])
+  })
+
+  it('returns empty array if overloaded checkSupport of component throws error', () => {
+    const componentRoot = document.createElement('div')
+    componentRoot.setAttribute('data-module', 'mock-component')
+    document.body.appendChild(componentRoot)
+
+    // Silence warnings in test output, and allow us to 'expect' them
+    jest.spyOn(global.console, 'log').mockImplementation()
+
+    const checkSupportMock = jest.fn(() => {
+      throw Error('Mock error')
+    })
+
+    class MockComponentWithCheckSupport extends MockComponent {
+      checkSupport() {
+        checkSupportMock()
+      }
+    }
+
+    const result = createAll(MockComponentWithCheckSupport)
+    expect(checkSupportMock).toHaveBeenCalled()
+    expect(result).toStrictEqual([])
+    expect(global.console.log).toHaveBeenCalledWith(expect.any(Error))
```

Comment:
**suggestion** Is there a way we can check this is the error thrown by the support mock (for example, comparing the text of the error)?